src/containers/SigninContainer/form.tsx

```diff
@@ -0,0 +1,168 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { Form, Field } from 'react-final-form';
+
+import { Typography } from '@/assets';
+import Pages from '@/constants/pages';
+import CInput from '@/components/CInput';
+import CButton from '@/components/CButton';
+import CCheckbox from '@/components/CCheckbox';
+import { FormValues, IApiError } from '@/constants/types';
+import { required, minLength, validateEmail, validatePassword } from '@/utils/validators';
+
+import FetchLogin from './fetchLogin';
+
+const SignUpForm = () => {
+  const routes = useRouter();
+  const [isRememberChecked, setIsRememberChecked] = useState(false);
+  const [resMessage, setResMessage] = useState({
+    status: '',
+    title: '',
+    message: '',
+  });
+
+  const handleRedirectToSignUp = () => {
+    routes.push(Pages.SIGNUP);
+  };
+
+  const handleRememberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsRememberChecked(e.target.checked);
+  };
+
+  const onSubmit = async (values: FormValues) => {
+    try {
+      const data = await FetchLogin({ email: values.email, password: values.password });
+      if (data.result?.token) {
+        setResMessage({
+          status: 'success',
+          title: '',
+          message: '',
+        });
+        if (isRememberChecked) {
+          localStorage.setItem('token', data.result?.token);
+        } else {
+          sessionStorage.setItem('token', data.result?.token);
+        }
+        routes.push(Pages.DASHBOARD);
+      }
+    } catch (error: IApiError | any) {
+      if (error.data.message === 'User not exist') {
+        setResMessage({
+          status: 'error',
+          title: 'Login Failed',
+          message: 'User with this Email not found, Sign up instead.',
+        });
+      } else if (error.data.message === 'Invalid credentials') {
+        setResMessage({
+          status: 'error',
+          title: 'Login Failed',
+          message: 'Email or password does not match with your information in our database.',
+        });
+      } else {
+        setResMessage({
+          status: 'error',
+          title: 'Login Failed',
+          message: 'An error occurred during authentication, try again.',
+        });
+        error = error.data.message;
+      }
+    }
+    return {};
+  };
+
+  const composeValidators =
```

Comment:
instead of having the same function in both signup and signin, make a new file and put it there and use in both places.